### ISSUE TITLE
Crate doesn't work for new target triples

### ISSUE THREAD

**Issue by posborne** [dcuddeback/serial-rs#33]
In my case, I am using the Yocto meta-rust layer to build some code that in turn depends on this crate.  Within this particular build system, the the target ends up being `arm-poky-linux-gnueabi`.  The serial crate fails to compile with the following error:

```
.../serial-0.3.4/src/posix/tty.rs:2:1: 2:22 error: can't find crate for `termios` [E0463]
.../cargo_home/registry/src/github.com-88ac128001ac3a9a/serial-0.3.4/src/posix/tty.rs:2 extern crate termios;
                                                                                        ^~~~~~~~~~~~~~~~~~~~~
```

The cause of this is that the Cargo.toml for this package seems to be trying to guess at all the possible targets that might exist that would require specific dependencies.

There are better approaches for this in more recent version of Cargo (http://doc.crates.io/specifying-dependencies.html#platform-specific-dependencies).  I'll open a PR shortly with an attempt to fix this issue.

**Comment by dcuddeback** [comment]
Hi. This is a known limitation in older versions of Cargo, which I discussed a little bit in #24 as well. The minimum supported version of Rust for serial 0.3.x is 1.3, which ships with a version of Cargo that doesn't support `cfg()` syntax. (It only recently reached the stable channel.) I think it's best if point releases don't break builds, so I'm waiting until v0.4 to adopt the `cfg()` syntax in Cargo.toml.

For the v0.3 branch, if you want to add additional targets to Cargo.toml, I'd be happy to publish a new point release.

**Comment by posborne** [comment]
Ah, thanks for the explanation and history.  Is there a branch with changes for 0.4 that exists today?  Since the target in my case is not standard, I can probably get by until 0.4 pointing Cargo at my fork.  It turns out that I am actually stuck on 1.7.0 until I land some other patches, so I might need some hacks anyway.

I can leave this open if you think you might want to include it for 0.4 (maybe merged into a different branch) or I can close it.  Either way works for me.

**Comment by dcuddeback** [comment]
@posborne No problem. I do have a branch that I've been trying out for v0.4, which already has the changes to Cargo.toml. We can leave this story open for visibility until v0.4 is merged.

> Since the target in my case is not standard

That's what I thought at first, too, but when I searched `arm-poky-linux-gnueabi`, it seems like it's pretty common for Yocto. I'd be okay adding it to Cargo.toml for now. It's just a temporary fix until I release v0.4.

**Comment by posborne** [comment]
I've got another workaround in place on a fork we are using temporarily.  That sufficies for me in the short term.  I would not be opposed to adding the triple either, however.
